<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <!-- ====================================================== -->
  <!-- Ignition ros2_control plugin -->
  <!-- ====================================================== -->
  <gazebo>
    <plugin
        filename="libign_ros2_control-system.so"
        name="ign_ros2_control::IgnitionROS2ControlPlugin">
      <parameters>$(find towing)/config/ros2_controllers.yaml</parameters>
    </plugin>
  </gazebo>

  <!-- ====================================================== -->
  <!-- ros2_control hardware definition -->
  <!-- ====================================================== -->
  <ros2_control name="IgnitionSystem" type="system">

    <hardware>
      <plugin>ign_ros2_control/IgnitionSystem</plugin>
    </hardware>

    <!-- ================= DRIVE WHEEL ================= -->
    <joint name="rear_left_wheel_joint">
      <command_interface name="velocity"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>

    <joint name="rear_right_wheel_joint">
      <command_interface name="velocity"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>

    <!-- ================= STEERING ================= -->
    <joint name="steer_joint">
      <command_interface name="position"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>

  </ros2_control>

</robot>
